core/shared/src/main/scala/scaladex/core/api/AutocompletionEndpoints.scala

```diff
@@ -0,0 +1,64 @@
+package scaladex.core.api
+
+import endpoints4s.algebra
+
+case class AutocompletionRequest(
+    query: String,
+    you: Boolean,
+    topics: Seq[String],
+    targetTypes: Seq[String],
+    scalaVersions: Seq[String],
+    scalaJsVersions: Seq[String],
+    scalaNativeVersions: Seq[String],
+    sbtVersions: Seq[String],
+    contributingSearch: Boolean
+)
+
+case class AutocompletionResponse(
+    organization: String,
+    repository: String,
+    description: String
+)
+
+// Autocompletion endpoints are implemented by the server and invoked by the web client.
+// There definition is shared here, for consistency.
+trait AutocompletionEndpoints extends algebra.Endpoints with algebra.JsonEntitiesFromSchemas {
+
+  /** Enrich a request with session information */
+  def withOptionalSession[A](request: Request[A]): Request[WithSession[A]]
+
+  /** A type `A` enriched with session information (unknown yet, defined by the web client and server) */
+  type WithSession[A]
+
+  // JSON schema of the autocompletion response entity
+  implicit val autocompletionResponseSchema: JsonSchema[AutocompletionResponse] =
```

Comment:
Out of curiosity, is it possible to derive a `JsonSchema` from a circe `Codec`?

Reply:
No, because Circe codecs are not “introspectable” (they are just functions `Json => DecodingResult[A]`). But it is possible to derive a Circe codec from a `JsonSchema`, if needed.

Reply:
I guess not because the documentation of the schema would be missing. But I see that we can do the other way around: derive a `Codec` from a `JsonSchema` (https://endpoints4s.github.io//interpreters/circe.html).